catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java

```diff
@@ -518,77 +498,44 @@ private void deleteListOfMetacards(
-      throws IngestException {
-
-    for (SolrDocument doc : docs) {
-      if (LOGGER.isDebugEnabled()) {
-        LOGGER.debug("SOLR DOC: {}", doc.getFieldValue(Metacard.ID + SchemaFields.TEXT_SUFFIX));
+  private List<Metacard> getMetacards(
+      List<? extends Serializable> identifierPaged, String fieldName) throws IngestException {
+    if (fieldName.equals(Metacard.ID + SchemaFields.TEXT_SUFFIX)) {
+      Set<String> ids =
+          identifierPaged
+              .stream()
+              .filter(Objects::nonNull)
+              .map(Object::toString)
+              .collect(Collectors.toSet());
+      try {
+        return client.getIds(ids);
+      } catch (UnsupportedQueryException e) {
+        LOGGER.info("Failed to get list of Solr documents by ID for delete.", e);
+        throw new IngestException(COULD_NOT_COMPLETE_DELETE_REQUEST_MESSAGE);
       }
-
+    } else {
       try {
-        deletedMetacards.add(client.createMetacard(doc));
-      } catch (MetacardCreationException e) {
-        LOGGER.info("Metacard creation exception creating metacards during delete", e);
+        return client.query(client.getIdentifierQuery(fieldName, identifierPaged));
+      } catch (UnsupportedQueryException e) {
+        LOGGER.info("Failed to get list of Solr documents for delete.", e);
         throw new IngestException(COULD_NOT_COMPLETE_DELETE_REQUEST_MESSAGE);
       }
     }
   }
 
-  private void addPendingNrtDeletedMetacards(
-      List<Metacard> deletedMetacards, List<? extends Serializable> identifiers) {
-    if (deletedMetacards.size() < identifiers.size()
-        && pendingNrtIndex.getAllPresent(identifiers).size() > 0) {
-      Map<String, Metacard> matches = pendingNrtIndex.getAllPresent(identifiers);
-
-      deletedMetacards.forEach(m -> matches.remove(m.getId()));
-      deletedMetacards.addAll(matches.values());
-    }
-  }
-
-  private Metacard copyMetacard(Metacard original) {
-    MetacardImpl copy = new MetacardImpl(original, original.getMetacardType());
-    copy.setSourceId(original.getSourceId());
-    return copy;
-  }
-
-  private SolrDocumentList getSolrDocumentList(
-      List<? extends Serializable> identifierPaged, String fieldName) throws IngestException {
-    SolrQuery query = new SolrQuery(client.getIdentifierQuery(fieldName, identifierPaged));
-    query.setRows(identifierPaged.size());
-
-    QueryResponse solrResponse;
-    try {
-      solrResponse = solr.query(query, METHOD.POST);
-    } catch (SolrServerException | SolrException | IOException e) {
-      LOGGER.info("Failed to get list of Solr documents for delete.", e);
-      throw new IngestException(COULD_NOT_COMPLETE_DELETE_REQUEST_MESSAGE);
-    }
-    return solrResponse.getResults();
-  }
-
-  private String getQuery(String attributeName, List<String> ids) throws IngestException {
-
-    StringBuilder queryBuilder = new StringBuilder();
+  private String getQuery(String attributeName, Set<String> ids) throws IngestException {
```

Comment:
✏️ Is it possible to disambiguate this name a bit more? It looks like it gets a specific kind of query.

Reply:
The query is based on the parameters of attribute and IDs for that attribute.  What kind of query is returned is an implementation detail.